### PR TITLE
Redux in WBooks Application

### PR DESCRIPTION
**Pull request by LoadCode**
## Summary
This pull request introduces Redux and React-Redux to the state management of the WBooks application.
In this PR we define the dependencies for a basic redux implementation: define storage, reducers, actions, etc.

## Screenshots

Bellow are the screenshots of the application working on different devices (Android & iOS)

#### Galaxy S
galaxy_s

#### Pixel 2
pixel2

#### Pixel 3XL
pixel3xl

#### iPhone SE
iphone-se

#### iPhone 8
iphone-8

#### iPhone 11
iphone-11

## Trello Card
[(Nº 5.1) Redux I - Actions, Reducers, Store](https://trello.com/c/w4PqfjAm/25-n-51-redux-i-actions-reducers-store)

### CODE REVIEW

File: wbooks/src/interfaces/AppInfo.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { IBook } from '@interfaces/BookInfo';
+
+export interface AppState {
+  bookList: IBook[];
+  idSelectedBook: number | null;
```
Comment (sfernandez11) [inline wbooks/src/interfaces/AppInfo.ts:5]: Why this? I think it's not necessary

File: wbooks/src/interfaces/BookInfo.ts [hunk 2]
```diff
@@ -7,3 +7,8 @@ export interface IBook {
   year: string;
   imageUrl: string | null;
 }
+
+export interface IBookAction {
+  type: string;
+  id?: number;
```
Comment (sfernandez11) [inline wbooks/src/interfaces/BookInfo.ts:13]: Why this? I think it's not necessary:
```
export interface IBookAction {
  type: string;
  payload: IBook[];
}
```

File: wbooks/src/redux/books/actions.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { IBookAction } from '@interfaces/BookInfo';
+
+export const actionNames = {
+  getBookList: 'GET_BOOK_LIST',
+  getBookById: 'GET_BOOK_BY_ID'
```
Comment (sfernandez11) [inline wbooks/src/redux/books/actions.ts:5]: Why this? I think it's not necessary

File: wbooks/src/redux/books/actions.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+};
+
+export const actionCreators = {
+  getBookList: (): IBookAction => ({
+    type: actionNames.getBookList
+  }),
+  getBookById: (id: number): IBookAction => ({
+    type: actionNames.getBookById,
+    id
+  })
```
Comment (sfernandez11) [inline wbooks/src/redux/books/actions.ts:15]: Why this? I think it's not necessary

File: wbooks/src/redux/books/actions.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+export const actionNames = {
+  getBookList: 'GET_BOOK_LIST',
+  getBookById: 'GET_BOOK_BY_ID'
+};
+
+export const actionCreators = {
+  getBookList: (): IBookAction => ({
+    type: actionNames.getBookList
+  }),
```
Comment (sfernandez11) [inline wbooks/src/redux/books/actions.ts:11]: You must send the `payload` from the action to store in the reducer:
```
  getBookList: (): IBookAction => ({
    type: actionNames.getBookList,
    payload: BOOKS_MOCK
  }),
```

File: wbooks/src/redux/books/reducers.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import { BOOKS_MOCK } from '@constants/mockBooks';
+import { actionNames as bookActionNames } from '@redux/books/actions';
+import { AppState } from '@interfaces/AppInfo';
+import { IBookAction } from '@interfaces/BookInfo';
+
+const initialState: AppState = {
+  bookList: [],
+  idSelectedBook: null
```
Comment (sfernandez11) [inline wbooks/src/redux/books/reducers.ts:8]: Why this? I think it's not necessary

File: wbooks/src/redux/books/actions.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import { IBookAction } from '@interfaces/BookInfo';
+
+export const actionNames = {
+  getBookList: 'GET_BOOK_LIST',
```
Comment (sfernandez11) [inline wbooks/src/redux/books/actions.ts:4]: `GET_BOOK_LIST: 'GET_BOOK_LIST',`

File: wbooks/src/redux/books/reducers.ts [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+};
+
+const reducerBooks = (state = initialState, action: IBookAction): AppState => {
+  switch (action.type) {
+    case bookActionNames.getBookList: {
```
Comment (sfernandez11) [inline wbooks/src/redux/books/reducers.ts:13]: `bookActionNames.GET_BOOK_LIST`

File: wbooks/src/redux/books/reducers.ts [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+      return {
+        ...state,
+        bookList: BOOKS_MOCK
```
Comment (sfernandez11) [inline wbooks/src/redux/books/reducers.ts:16]: `bookList: action.payload`

File: wbooks/src/redux/books/reducers.ts [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+      };
+    }
+    default:
+      return state;
+  }
+};
+
+export { reducerBooks };
```
Comment (sfernandez11) [inline wbooks/src/redux/books/reducers.ts:24]: `export default reducerBooks;`

File: wbooks/src/redux/books/reducers.ts [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+import { actionNames as bookActionNames } from '@redux/books/actions';
+import { AppState } from '@interfaces/AppInfo';
+import { IBookAction } from '@interfaces/BookInfo';
+
+const initialState: AppState = {
+  bookList: [],
+  idSelectedBook: null
+};
+
+const reducerBooks = (state = initialState, action: IBookAction): AppState => {
```
Comment (sfernandez11) [inline wbooks/src/redux/books/reducers.ts:11]: Note: We use the reducer only for store variables, all the logic must be in the actions

File: wbooks/src/redux/store.ts [hunk 12]
```diff
@@ -0,0 +1,7 @@
+import { createStore } from 'redux';
+
+import { reducerBooks } from './books/reducers';
```
Comment (sfernandez11) [inline wbooks/src/redux/store.ts:3]: `import reducerBooks from './books/reducers';`

**Review by sfernandez11 — COMMENTED** [review]
Great work Julio, some comments...

**Review by sfernandez11 — APPROVED** [review]
🥇

File: wbooks/src/interfaces/bookInfo.ts [hunk 13]
```diff
@@ -7,3 +7,8 @@ export interface IBook {
   year: string;
   imageUrl: string | null;
 }
+
+export interface IBookAction {
+  type: string;
+  payload: any;
```
[the remark this answers is not in the document]
Comment (LoadCode) [reply wbooks/src/interfaces/bookInfo.ts:13]: You're right, I'll fix it

**Review by sfernandez11 — APPROVED** [review]
🥇

**Pull request merged**